libnm-core/nm-setting-tc-config.c

```diff
@@ -62,11 +62,19 @@ nm_tc_qdisc_new (const char *kind,
 {
 	NMTCQdisc *qdisc;
 
+	if (!kind || !*kind) {
+		g_set_error (error,
+		             NM_CONNECTION_ERROR,
+		             NM_CONNECTION_ERROR_INVALID_PROPERTY,
+		             _("kind is missing."));
+		return NULL;
+	}
+
 	if (!kind || !*kind || strchr (kind, ' ') || strchr (kind, '\t')) {
```

Comment:
Minor note: here we can now remove the check on "!kind" and "!*kind" as already checked above:
`if (strchr (kind, ' ') || strchr (kind, '\t'))`